spec/openhab/core/types/time_series_spec.rb

```diff
@@ -49,6 +49,26 @@
       ts.add(Time.at(0), DecimalType.new(1))
       expect(ts.begin).to eql Instant.of_epoch_second(0)
     end
+
+    it "accepts a string value" do
+      ts.add(Time.at(0), "1")
+      expect(ts.first.state.class).to be StringType
+      expect(ts.first.state).to eq StringType.new("1")
```

Comment:
Get rid of the previous line, and change this to eql (comparison without type change)

Reply:
That's what I tried before, but I got this:

```
  1) Java::OrgOpenhabCoreTypes::TimeSeries#add accepts a string value
     Failure/Error: to_s.compare_to(other.to_s).zero?
     
     NoMethodError:
       undefined method `compare_to' for "1":String
       Did you mean?  Complex
     # ./lib/openhab/core/types/string_type.rb:29:in `eql?'
     # ./spec/openhab/core/types/time_series_spec.rb:56:in `block in <main>'
```

Reply:
Should we change `String#eql?` from

```
        def eql?(other)
          return false unless other.instance_of?(self.class)

          to_s == other.to_s
        end
```

to
```
          to_s == other.to_s
```

Reply:
I've changed StringType#eql? now